src/burnchains/events.rs

```diff
@@ -89,6 +101,28 @@ where
     ClarityValue::try_deserialize_hex_untyped(&str_val).map_err(DeserError::custom)
 }
 
+/// Convert a slice of u8 to a hex string. TODO: use general fn.
+pub fn to_hex(s: &[u8]) -> String {
+    let mut r = String::with_capacity(s.len() * 2);
+    for b in s.iter() {
+        write!(r, "{:02x}", b).unwrap();
+    }
+    return r;
+}
+
+/// Serialize a clarity value to work with `deser_clarity_value`.
+fn ser_clarity_value<S>(value: &ClarityValue, s: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut byte_serialization = Vec::new();
+    value
+        .serialize_write(&mut byte_serialization)
+        .expect("IOError filling byte buffer.");
+    let string_value = to_hex(byte_serialization.as_slice());
+    s.serialize_str(&string_value)
```

Comment:
`StacksMessageCodec` auto-implements a `serialize_to_vec()` function that helps cut down on some boiler plate when serializing to a vector:

```suggestion
    let byte_serialization = value.serialize_to_vec();
    let string_value = to_hex(byte_serialization.as_slice());
    s.serialize_str(&string_value)
```